### PR TITLE
Update Gradle, CircleCI, Makefile in Prep for Places 3.0.0 release

### PR DESCRIPTION
**Pull request by PravinPK**

### CODE REVIEW

File: .github/workflows/maven-snapshot.yml
Comment (spoorthipujariadobe) [inline .github/workflows/maven-snapshot.yml:1]: Can we also update the `maven-release.yml` file?

File: .github/workflows/update-version.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    inputs:
+      version: 
+        description: 'New version to use for the Places extension. Example: 3.0.0'
+        required: true
+
+      core-dependency:
+        description: '[Optional] Update Core dependency in pom.xml. Example: 3.0.0'
+        required: false
+
+       edge-dependency:
```
Comment (spoorthipujariadobe) [inline .github/workflows/update-version.yml:13]: Does Places have a hard dependency on Edge? Why do we need this?

File: .github/workflows/update-version.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        description: '[Optional] Update Places dependency in pom.xml. Example: 3.0.0'
```
Comment (spoorthipujariadobe) [inline .github/workflows/update-version.yml:14]: `Update Places dependency` -> `Update Edge dependency`

File: .github/workflows/update-version.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+on:
+  workflow_dispatch:
+    inputs:
+      version: 
+        description: 'New version to use for the Places extension. Example: 3.0.0'
+        required: true
+
+      core-dependency:
+        description: '[Optional] Update Core dependency in pom.xml. Example: 3.0.0'
+        required: false
```
Comment (spoorthipujariadobe) [inline .github/workflows/update-version.yml:11]: Should it be `required: true` since Places has hard dependency on Core?

Comment (spoorthipujariadobe) [reply .github/workflows/update-version.yml:11]: Also let's remove `[Optional]`

File: Makefile [hunk 4]
```diff
@@ -1,46 +1,56 @@
...
-		(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) checkstyle)
+init:
+	git config core.hooksPath .githooks
 
 clean:
-	  (./code/gradlew -p code clean)
+	(./code/gradlew -p code clean)
+
+format:
+	(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) spotlessApply)
```
Comment (spoorthipujariadobe) [inline Makefile:11]: Let's also apply spotless to the `TEST-APP-FOLDER-NAME`

File: Makefile [hunk 5]
```diff
@@ -1,46 +1,56 @@
...
-		(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) publishReleasePublicationToSonatypeRepository)
+	(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) javadocJar)
 
 assemble-phone:
-		(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) assemblePhone)
+	(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) assemblePhone)
 		
 assemble-phone-release:
-		(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) assemblePhoneRelease)
+	(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) assemblePhoneRelease)
```
Comment (spoorthipujariadobe) [inline Makefile:44]: We're missing `assemble-phone-debug` here

File: Makefile [hunk 6]
```diff
@@ -1,46 +1,56 @@
...
 
-ci-publish-staging: clean assemble-phone-release
-		(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) publishReleasePublicationToSonatypeRepository --stacktrace)
+ci-publish-staging:
+	(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) publishReleasePublicationToSonatypeRepository --stacktrace)
 
-ci-publish-main: clean assemble-phone-release
-		(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) publishReleasePublicationToSonatypeRepository -Prelease)
\ No newline at end of file
+ci-publish-main:
```
Comment (spoorthipujariadobe) [inline Makefile:55]: Can we rename this to `ci-publish` in keeping with the other extensions?

File: .github/workflows/update-version.yml (hunk 1 above)
Comment (spoorthipujariadobe) [inline .github/workflows/update-version.yml:13]: There is an extra space here before `edge-dependency` which will make the workflow fail

File: code/gradle.properties [hunk 7]
```diff
@@ -11,19 +11,19 @@
...
 mavenRepoName=AdobeMobilePlacesSdk
 mavenRepoDescription=Android Places Extension for Adobe Mobile Marketing
 mavenUploadDryRunFlag=false
-mavenCoreVersion=2.3.0
 
-android.disableAutomaticComponentCreation=true
-android.useAndroidX=true
+# production versions for production build
+mavenCoreVersion=3.0.0
+mavenPlacesVersion=2.0.0
```
Comment (spoorthipujariadobe) [inline code/gradle.properties:28]: Why do we have `mavenPlacesVersion` here? Is this supposed to be `mavenEdgeVersion` instead?

File: code/places/build.gradle.kts [hunk 8]
```diff
@@ -0,0 +1,32 @@
+plugins {
+    id("aep-library")
```
Comment (spoorthipujariadobe) [inline code/places/build.gradle.kts:2]: Do we also need `id("com.android.library")` and `id("org.jetbrains.kotlin-android")`? I don't see this in the `aep-library`

File: code/places/build.gradle.kts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+val mavenCoreVersion: String by project
+val mavenPlacesVersion: String by project
```
Comment (spoorthipujariadobe) [inline code/places/build.gradle.kts:6]: Why do we need this?

File: code/places/build.gradle.kts [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    enableSpotless = true
+ 
+    publishing {
+        gitRepoName = "aepsdk-places-android"
+        addCoreDependency(mavenCoreVersion)
+    }
+}
+
+dependencies {
+    implementation("com.adobe.marketing.mobile:core:$mavenCoreVersion-SNAPSHOT")
```
Comment (spoorthipujariadobe) [inline code/places/build.gradle.kts:20]: We're missing edge dependency here

File: code/places/build.gradle.kts [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    implementation ("com.google.android.gms:play-services-location:21.1.0")
+
+    // testImplementation dependencies provided by aep-library:
+    // MOCKITO_CORE, MOCKITO_INLINE, JSON
+
+    testImplementation ("com.fasterxml.jackson.core:jackson-databind:2.12.7")
```
Comment (spoorthipujariadobe) [inline code/places/build.gradle.kts:26]: Are we using this anywhere in our tests? Let's remove it if we aren't

File: code/places/build.gradle.kts [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+    // androidTestImplementation dependencies provided by aep-library:
+    // ANDROIDX_TEST_EXT_JUNIT, ESPRESSO_CORE
+
+    androidTestImplementation ("com.fasterxml.jackson.core:jackson-databind:2.12.7")
```
Comment (spoorthipujariadobe) [inline code/places/build.gradle.kts:31]: Let's check if we are using this and remove if not needed

File: code/testapp/build.gradle.kts [hunk 13]
```diff
@@ -0,0 +1,37 @@
+import com.adobe.marketing.mobile.gradle.BuildConstants
+
+plugins {
+    id("com.android.application")
+}
+
+android {
+      namespace = "com.adobe.placestestapp"
```
Comment (cacheung) [inline code/testapp/build.gradle.kts:8]: According to the wiki, "Refactor Test application to separate package: If your project's test application doesn't have a unique namespace, move it to a namespace like com.adobe.marketing.mobile.<extension>.testapp for clearer separation. Make sure the app name follows similar naming (ex UserProfile Test App)."

Not sure if you want to refactor to 
com.adobe.marketing.mobile.places.testapp

File: .github/workflows/maven-snapshot.yml [hunk 14]
```diff
@@ -10,18 +10,20 @@
 # governing permissions and limitations under the License.
 #
 
-name: Publish package to the Maven Central Snapshot Repository
-on: workflow_dispatch
-
+name: Publish Snapshot
+on:
+  push:
```
Comment (cacheung) [inline .github/workflows/maven-snapshot.yml:15]: We updated to `workflow_dispatch:` [here](https://github.com/adobe/aepsdk-userprofile-android/blob/dev-v3.0.0/.github/workflows/maven-snapshot.yml#L3).

So won't automatically running on push to staging branch.

File: .github/workflows/update-version.yml (hunk 2 above)
Comment (PravinPK) [reply .github/workflows/update-version.yml:14]: Places doesn't have a hard dependency on Edge..
I will remove this part

File: .github/workflows/update-version.yml (hunk 3 above)
Comment (PravinPK) [reply .github/workflows/update-version.yml:11]: Agreed this field is not mandatory. Core version dependency can remain the same despite the update to Places version 
 Keeping them the same way as other extension
 https://github.com/adobe/aepsdk-userprofile-android/blob/main/.github/workflows/update-version.yml#L10-L12

File: Makefile (hunk 4 above)
Comment (PravinPK) [reply Makefile:11]: Good call

File: Makefile (hunk 5 above)
Comment (PravinPK) [reply Makefile:44]: Nice catch

File: .github/workflows/update-version.yml (hunk 1 above)
Comment (PravinPK) [reply .github/workflows/update-version.yml:13]: Its a soft dependency and not a hard dependency. WIll remove them
Thanks

File: .circleci/config.yml [hunk 15]
```diff
@@ -12,79 +12,130 @@
...
+      - functional-test:
+          requires:
+            - validate-code
  
 jobs:
+  validate-code:
+    executor:
+      name: android/android-docker
+      resource-class: medium
+      tag: 2024.01.1
```
[the remark this answers is not in the document]
Comment (PravinPK) [reply .circleci/config.yml:36]: Thank you for the heads up

File: .github/workflows/maven-snapshot.yml (hunk 14 above)
Comment (PravinPK) [reply .github/workflows/maven-snapshot.yml:15]: 👍🏼  Sounds good

File: code/testapp/build.gradle.kts (hunk 13 above)
Comment (PravinPK) [reply code/testapp/build.gradle.kts:8]: I will refactor this in my next PR

File: .github/workflows/update-version.yml [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+
+jobs:
+  update-version:
+    runs-on: ubuntu-latest
+
+    steps:    
+    - name: Checkout
+      uses: actions/checkout@v3
+    
+    - name: Update Places Version
```
[the remark this answers is not in the document]
Comment (PravinPK) [reply .github/workflows/update-version.yml:25]: Edge is not a hard dependency. Sorry for misleading

File: .circleci/config.yml [hunk 17]
```diff
@@ -12,79 +12,130 @@
...
-            done
+          name: Assemble Phone
+          command: make assemble-phone
+
       - run:
-          name: Run Unit Tests
-          command: make unit-test-coverage
-      - store_test_results:
-          path: code/android-places-library/build/test-results/testPhoneDebugUnitTest
+          name: Assemble for Jitpack
```
[the remark this answers is not in the document]
Comment (PravinPK) [reply .circleci/config.yml:85]: JItpack publish for places is unnecessary as well. Removing this step

File: code/places/build.gradle.kts (hunk 10 above)
Comment (PravinPK) [reply code/places/build.gradle.kts:20]: Places does not have hard dependencies on edge. We can ignore them here

File: code/places/build.gradle.kts (hunk 12 above)
Comment (PravinPK) [reply code/places/build.gradle.kts:31]: Copy pasta problem.. Good to be removed

File: code/places/build.gradle.kts (hunk 9 above)
Comment (PravinPK) [reply code/places/build.gradle.kts:6]: This picks the value from gradle.properties

We dont need the mavenPlacesVersion variable in this gradle file. Removing them.
Thanks for pointing out

File: code/gradle.properties (hunk 7 above)
Comment (PravinPK) [reply code/gradle.properties:28]: Removed mavenPlacesVersion. It doesn't make sense here.
And also mavenEdgeVersion is not a hard dependency on places will ignore that one too

File: .github/workflows/maven-snapshot.yml [hunk 18]
```diff
@@ -10,18 +10,20 @@
...
 jobs:
   publish:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Set up Java
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
```
[the remark this answers is not in the document]
Comment (PravinPK) [reply .github/workflows/maven-snapshot.yml:23]: Haven't heard of this. This was new to me
Thanks for the mentioning Calise

**Review by spoorthipujariadobe — APPROVED** [review]
LGTM

File: .github/workflows/maven-snapshot.yml (hunk 18 above)
[the remark this answers is not in the document]
Comment (cacheung) [reply .github/workflows/maven-snapshot.yml:23]: userprofile is doing the same
https://github.com/adobe/aepsdk-userprofile-android/blob/dev-v3.0.0/.github/workflows/maven-snapshot.yml#L14

Are you updating it for Places?

File: .github/workflows/maven-release.yml [hunk 19]
```diff
@@ -10,36 +10,68 @@
...
+    inputs:
+      tag:
+        description: 'tag/version'
+        required: true        
 
+      action_tag:
+        description: 'Create tag? ("no" to skip)'
+        required: true
+        default: 'yes'
 jobs:
```
Comment (cacheung) [inline .github/workflows/maven-release.yml:25]: Userprofile update some changes for its yml, now with no ref:main and added distribution: 'zulu'.  To make it consistent, see if you want to do the same for places.

jobs:
  publish:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3

      - name: Set up Java
        uses: actions/setup-java@v3
        with:
          distribution: 'zulu'
          java-version: 17

Comment (PravinPK) [reply .github/workflows/maven-release.yml:25]: Got that in

File: code/testapp/build.gradle.kts [hunk 20]
```diff
@@ -0,0 +1,37 @@
...
+dependencies {
+    implementation(project(":places"))
+    implementation("com.adobe.marketing.mobile:core:2.+")
+    implementation("com.adobe.marketing.mobile:assurance:2.+")
+    implementation("com.adobe.marketing.mobile:edge:2.+")
+
+    implementation("com.google.android.gms:play-services-location:21.1.0")
+    implementation("com.google.android.gms:play-services-maps:18.2.0")
+
+    implementation("androidx.appcompat:appcompat:1.5.1")
```
Comment (cacheung) [inline code/testapp/build.gradle.kts:35]: We have BuildConstants for AppCompat, you might want to use it.
implementation("androidx.appcompat:appcompat:${BuildConstants.Versions.ANDROIDX_APPCOMPAT}")

Comment (cacheung) [reply code/testapp/build.gradle.kts:35]: We don't need to do that in App dependencies actually.

**Pull request merged**